### DISCUSSION TITLE
First Attempt at Deepgram Live Connection Failure

### DISCUSSION THREAD

**Discussion by dnwint2** [deepgram/community#66] [General help]
Hello,
           I more or less copypasted the Deepgram proxy server sample code. I seem to have a problem where the first time I run the code, the server crashes with the following error: 

<img width="931" alt="ConnectionError" src="https://user-images.githubusercontent.com/55368487/208050193-8a144ac0-3871-4db9-b47c-d68552930837.png">

Thereafter, the server works properly. 

How can I avoid this initial problem + server failure? 

Thanks!

**Comment by BekahHW** [comment]
Can you share your code?

**Comment by dnwint2** [comment]
Absolutely. There are two scenarios (I got them out of order as I cloned the server.js file and set it up for each scenario)

1. server1.js & error1.png - Placed the deepgramLive expression and deepgram.live event handlers in the websocket callback so that I wouldn't have async issues with the deepgramLive cnxn dying (this is my hypothesis here - that it dies before the client.js file gets up and running with the websocket, i.e. before it gets its first message, or isn't created fast enough, one or the other.
2. server2.js & error2.png - this is the original scenario, the copy-paste of the original proxy server example code + additions for other stuff. This error is an even stranger one. I think it's receiving the opening event as a message (?!) from the deepgram connection, and so doesn't see the transcription object

My current hypothesis is that the deepgramLive code is not robustly designed on you all's end, with two conflicting design elements: A) a timeout on the connection if no messages arrive after it's created, and B) an unparseable first message that prevents the connection from being create-able in the same callback function where parsing occurs. A follow-on hypothesis is that this is not detected (mostly) because most of the time, the first message occurs in the 10s window after the connection is created before the timeout. You can get a data point for this by running you all's sample proxy server code but waiting  20s before selecting yes on  the 'getUserMedia' microphone prompt.

I am 100% open to being wrong. 


<img width="1403" alt="server1_a" src="https://user-images.githubusercontent.com/55368487/208226925-571863f0-5ad1-46a4-9ee0-cc4dbf1c1ef3.png">
<img width="1398" alt="server1_b" src="https://user-images.githubusercontent.
<img width="932" alt="error1" src="https://user-images.githubusercontent.com/55368487/208226933-72276328-7c33-4941-828b-a6a28fee3968.png">
<img width="915" alt="error2" src="https://user-images.githubusercontent.com/55368487/208226934-88dd63bd-4466-46f1-9517-d28620ce0411.png">
com/55368487/208226928-51b4939e-8e4b-4b65-a698-873a68f2a025.png">
<img width="1440" alt="server2_a" src="https://user-images.githubusercontent.com/55368487/208226930-6b05ec6d-dc6b-4d21-9a59-7039f8f21fd5.png">
<img width="1440" alt="server2_b" src="https://user-images.githubusercontent.com/55368487/208226931-03a31eac-ca02-4225-93c9-ea3b433822d9.png">

**Reply by dnwint2** [reply]
The server1_b.png did not show up properly. Here it is again.
<img width="1398" alt="server1_b" src="https://user-images.githubusercontent.com/55368487/208226985-024a1701-c91a-4d3a-a7d3-a515acfd1ff9.png">

**Reply by dnwint2** [reply]
error1 did not appear to make it into the above posts either. Here it is:
<img width="932" alt="error1" src="https://user-images.githubusercontent.com/55368487/208227041-f8271b9a-555d-44d7-aa7a-d84af6d55baa.png">

**Comment by BekahHW** [comment]
Can you provide a link to the `Deepgram proxy server sample code` you mentioned initially?
